.github/workflows/kurobako.yml

```diff
@@ -0,0 +1,96 @@
+name: kurobako
+
+# TODO: Discuss the trigger configuration
+on:
+  pull_request: {}
+
+jobs:
+  benchmark:
+
+    runs-on: ubuntu-18.04
+    container: docker://dvcorg/cml-py3:latest
+
+    if: github.repository == 'optuna/optuna'
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install gnuplot
+      run: |
+        sudo apt update
+        sudo apt -y install gnuplot
+
+    - name: Cache kurobako CLI
+      id: cache-kurobako
+      uses: actions/cache@v2
+      with:
+        path: ./kurobako
+        key: kurobako-0-2-6
+
+    - name: Download kurobako CLI
+      if: steps.cache-kurobako.outputs.cache-hit != 'true'
+      run: |
+        curl -L https://github.com/sile/kurobako/releases/download/0.2.6/kurobako-0.2.6.linux-amd64 -o kurobako
+        chmod +x kurobako
+        ./kurobako -h
+
+    - name: Cache hpobench dataset
+      id: cache-hpobench-dataset
+      uses: actions/cache@v2
+      with:
+        path: ./fcnet_tabular_benchmarks
+        key: hpobench-dataset
+
+    - name: Download hpobench dataset
+      if: steps.cache-hpobench-dataset.outputs.cache-hit != 'true'
+      run: |
+        wget http://ml4aad.org/wp-content/uploads/2019/01/fcnet_tabular_benchmarks.tar.gz
+        tar xf fcnet_tabular_benchmarks.tar.gz
+
+    - name: Cache nasbench dataset
+      id: cache-nasbench-dataset
+      uses: actions/cache@v2
+      with:
+        path: ./nasbench_full.bin
+        key: nasbench-dataset
+
+    # Ref: https://github.com/sile/kurobako/wiki/NASBench
+    - name: Download nasbench dataset
+      if: steps.cache-nasbench-dataset.outputs.cache-hit != 'true'
+      run: |
+        curl -L $(./kurobako dataset nasbench url) -o nasbench_full.tfrecord
+        ./kurobako dataset nasbench convert nasbench_full.tfrecord nasbench_full.bin
+
+    # TODO: Discuss the benchmark task-solver-pruner configuration
+    - name: Run Benchmark
+      run: |
+        ./kurobako solver random | tee solver.json
+        ./kurobako problem hpobench ./fcnet_tabular_benchmarks/fcnet_protein_structure_data.hdf5 | tee problem.json
+        ./kurobako studies --solvers $(cat solver.json) --problems $(cat problem.json) | ./kurobako run > report.json
+        cat report.json | ./kurobako report > report.md
+        cat report.json | ./kurobako plot curve --errorbar
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: benchmark-report
+        path: |
+          report.md
+          images/**/*.png
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: benchmark-report
+        path: |
+          report.md
+          images/**/*.png
+
+    - name: cml_run
+      env:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        for file in `find images -name '*.png'`
+        do
+          echo $file
+          cml-publish $file --md >> report.md
+        done
+        cml-send-comment report.md
```

Comment:
Is it able to stop the comment to git revision? All watchers seems to receive new notifications every time you pushed new commits.
https://github.com/optuna/optuna/commit/57a96c53e860fc818ff77ce3c2890368231c2b09.

FYI, [github-actions-kurobako](https://github.com/c-bata/github-actions-kurobako) create/update a PR comment instead of git revision's comment.

Reply:
I'll try `cml-send-check` instead later. Ref: https://github.com/iterative/cml#cml-functions

Reply:
Will you get notifications even if this job only gets triggered by master push and releases and drops comments on them?